convert2rhel/actions/conversion/preserve_only_rhel_kernel.py

```diff
@@ -0,0 +1,258 @@
+                    loggerinst.info("\n%s" % info_message)
+                    self.add_message(
+                        level="INFO",
+                        id="CONFLICT_OF_KERNELS",
+                        title="Conflict of installed kernel versions",
+                        description=info_message,
+                    )
+                    pkghandler.handle_no_newer_rhel_kernel_available()
+                    kernel_update_needed = True
+
+        if kernel_update_needed:
+            pkghandler.update_rhel_kernel()
+
+
+class VerifyRhelKernelInstalled(actions.Action):
+    id = "VERIFY_RHEL_KERNEL_INSTALLED"
+    dependencies = ("INSTALL_RHEL_KERNEL",)
+
+    def run(self):
+        """Verify that the RHEL kernel has been successfully installed and raise an ERROR if not"""
+        super(VerifyRhelKernelInstalled, self).run()
+
+        loggerinst.info("Verifying that RHEL kernel has been installed")
+        installed_rhel_kernels = pkghandler.get_installed_pkgs_by_fingerprint(
+            system_info.fingerprints_rhel, name="kernel"
+        )
+        if len(installed_rhel_kernels) <= 0:
+            self.set_result(
+                level="ERROR",
+                id="NO_RHEL_KERNEL_INSTALLED",
+                title="No RHEL kernel installed",
+                description="There is no RHEL kernel installed on the system.",
+                remediations="Verify that the repository used for installing kernel contains RHEL packages.",
+            )
+            return
+
+        loggerinst.info("RHEL kernel has been verified to be on the system.")
+        self.add_message(
+            level="INFO",
+            id="RHEL_KERNEL_INSTALL_VERIFIED",
+            title="RHEL kernel install verified",
+            description="The RHEL kernel has been verified to be on the system.",
+        )
+
+
+class FixInvalidGrub2Entries(actions.Action):
+    id = "FIX_INVALID_GRUB2_ENTRIES"
+    dependencies = ("KERNEL_PACKAGES_INSTALLATION",)
+
+    def run(self):
+        """
+        On systems derived from RHEL 8 and later, /etc/machine-id is being used to identify grub2 boot loader entries per
+        the Boot Loader Specification.
+        However, at the time of executing convert2rhel, the current machine-id can be different from the machine-id from the
+        time when the kernels were installed. If that happens:
+        - convert2rhel installs the RHEL kernel, but it's not set as default
+        - convert2rhel removes the original OS kernels, but for these the boot entries are not removed
+        The solution handled by this function is to remove the non-functioning boot entries upon the removal of the original
+        OS kernels, and set the RHEL kernel as default.
+        """
+        super(FixInvalidGrub2Entries, self).run()
+
+        if system_info.version.major < 8:
+            # Applicable only on systems derived from RHEL 8 and later, and systems using GRUB2 (s390x uses zipl)
+            return
+
```

Comment:
Never tested. As it relates to a major version we should make sure this branch is covered